vint_test.go

```diff
@@ -41,10 +41,17 @@ var vints = []struct {
 
 func TestVInt(t *testing.T) {
 	for _, spec := range vints {
-		t.Run(strconv.FormatInt(spec.expected, 10), func(t *testing.T) {
-			res, err := ReadVInt(bytes.NewBuffer(spec.b))
+		t.Run(strconv.FormatInt(spec.number, 10), func(t *testing.T) {
+			// check that we can read the VInt and it's the correct value
+			res, err := ReadVInt(bytes.NewBuffer(spec.bytes))
 			assert.NoError(t, err, "ReadVInt should return successfully")
-			assert.Equal(t, spec.expected, res, "ReadVInt should return the correct result")
+			assert.Equal(t, spec.number, res, "ReadVInt should return the correct result")
+
+			// check that we can write the VInt and it's the correct value
```

Comment:
Let's make this a separate `t.Run` below (so it's a separate individual test). For the read it can be 

```go
t.Run(fmt.Sprintf("read %d", spec.number) ...
```

And then the write can be

```go
t.Run(fmt.Sprintf("write %d", spec.number) ...
```